File: gateOne/day 7/javascript/phonebook.js
const prompt = require("prompt-sync")();
let firstName = [];
let lastName = [];
let phoneNo = [];

function addContact(){
	let userFirstName = prompt("Enter first name: ");
	firstName.push(userFirstName); 
	let userLastName = prompt("Enter last name: ");
	lastName.push(userLastName);
	let userPhoneNo = prompt("Enter phone number: ");
	phoneNo.push(userPhoneNo);  
	console.log(">>>>>>>>> Contact saved successfully <<<<<<<<<<<");
	let desicion = prompt("Do you want to continue? ");
	if(desicion == "yes"){
		menu();
	}
}

function findContactByPhoneNo(){
	let phoneNumber = prompt("Enter mobile number: ");
	for(let element = 0; element < phoneNo.length; element++){
		if(phoneNumber == phoneNo[element]){
			console.log(firstName[element] + " " + lastName[element] + "\n" + phoneNo[element]);
		} 
	}
	let desicion2 = prompt("Do you want to continue? ");
	if(desicion2 == "yes"){
		menu();
	}
}

function findContactByFirstName(){
	let userFirstName = prompt("Enter first name: ");
	for(let index = 0; index < firstName.length; index++){
		if(userFirstName == firstName[index]){
			console.log(firstName[index] + " " + lastName[index] + "\n" + phoneNo[index]);
		} 
	}
	let desicion2 = prompt("Do you want to continue? ");
	if(desicion2 == "yes"){
		menu();
	}
}

function findContactByLastName(){
	let userLastName = prompt("Enter last name: ");
	for(let index1 = 0; index1 < lastName.length; index1++){
		if(userLastName == lastName[index1]){
			console.log(firstName[index1] + " " + lastName[index1] + "\n" + phoneNo[index1]);
		} 
	}
	let desicion2 = prompt("Do you want to continue? ");
	if(desicion2 == "yes"){
		menu();
	}
}


function editContact(){
	let contactName = prompt("Enter contact first name: ");
	for(let index2 = 0; index2 < firstName.length; index2++){
		if(contactName == firstName[index2]){
			console.log(firstName[index2] + " " + lastName[index2]);
			console.log("");
			console.log("Edit contact");
			let newFirstName = prompt("Enter new first name: ");
			firstName[index2] = newFirstName;
			
			let newLastName = prompt("Enter new last name: ");
			lastName[index2] = newLastName;

			let newPhoneNumber = prompt("Enter new phone number: ");
			phoneNo[index2] = newPhoneNumber;
		}
	}
	let desicion2 = prompt("Do you want to continue? ");
	if(desicion2 == "yes"){
		menu();
	}
}


function menu(){
	console.log("CHOOSE PERFERRED OPTION","\n", "1 -> Add contact","\n", "2 -> Remove contact", "\n", "3 -> Find contact by phone number","\n", "4 -> Find contact by first name","\n", "5 -> Find contact by last name","\n", "6 -> Edit contact","\n", "7 -> Back to menu");

let select = Number(prompt("Enter preferred number: "));
switch(select){
	case 1: 
		addContact();
		break;
	case 2: 
		console.log("remove contact");
		break;

	case 3: 
		findContactByPhoneNo();
		break;
	case 4: 
		findContactByFirstName();
		break;
	case 5: 
		findContactByLastName();
		break;
	case 6: 
		editContact();
		break;
	case 6: 
		menu();
		break;
}
}


console.log("CHOOSE PERFERRED OPTION","\n", "1 -> Add contact","\n", "2 -> Remove contact", "\n", "3 -> Find contact by phone number","\n", "4 -> Find contact by first name","\n", "5 -> Find contact by last name","\n", "6 -> Edit contact","\n", "7 -> Back to menu");

let select = Number(prompt("Enter preferred number: "));
switch(select){
	case 1: 
		addContact();
		break;
	case 2: 
		console.log("remove contact");
		break;

	case 3: 
		findContactByPhoneNo();
		break;
	case 4: 
		findContactByFirstName();
		break;
	case 5: 
		findContactByLastName();
		break;
	case 6: 
		editContact();
		break;
	case 6: 
		menu();
		break;
}

File: gateOne/day 7/python/phonebook.py
first_name = []
last_name = []
phone_no = []


def add_contact():
	user_first_name = str(input("Enter first name: "))
	first_name.append(user_first_name)
	user_last_name = str(input("Enter last name: "))
	last_name.append(user_last_name)
	phonenumber = str(input("Enter mobile number: "))
	phone_no.append(phonenumber)
	print(">>>>>>>>>> Contact saved successfully <<<<<<<<<<<<")
	decision = str(input("Add another contact? "))
	while decision.lower() == "yes":
		user_first_name = str(input("Enter first name: "))
		first_name.append(user_first_name)
		user_last_name = str(input("Enter last name: "))
		last_name.append(user_last_name)
		phonenumber = str(input("Enter mobile number: "))
		phone_no.append(phonenumber)
		print(">>>>>>>>>> Contact saved successfully <<<<<<<<<<<<")
		decision = str(input("Do you want to add more contacts? "))
		if decision.lower() != "yes":
			break
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()


def findcontact_byphonenumber():
	count = 0
	phonenumber = str(input("Enter mobile number: "))
	for element in phone_no:
		if phonenumber == element:
			print(first_name[count], last_name[count],"\n",phone_no[count])
		count += 1
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()


def findcontact_byfisrtname():
	count = 0
	user_first_name = str(input("Enter first name: "))
	for element in first_name:
		if user_first_name == element:
			print(first_name[count], last_name[count],"\n",phone_no[count])
		count += 1
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()	
	

def findcontact_bylastname():
	count = 0
	user_last_name = str(input("Enter last name: "))
	for element in last_name:
		if user_last_name == element:
			print(first_name[count], last_name[count],"\n",phone_no[count])
		count += 1
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()


def edit_contact():
	count = 0
	contact_name = str(input("Enter contact first name: "))
	for element in first_name:
		if contact_name == element:
			print(first_name[count], last_name[count])
			print("")
			print("Edit contact")
			new_firstname = str(input("Enter new first name: "))
			first_name[count] = new_firstname
			new_lastname = str(input("Enter new last name: "))
			last_name[count] = new_lastname
			new_number = str(input("Enter new phone number: "))
			phone_no[count] = new_number
		count += 1
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()		

def menu():
	print("CHOOSE PERFERRED OPTION\n"
	"1 -> Add contact\n"
	"2 -> Remove contact\n"
	"3 -> Find contact by phone number\n"
	"4 -> Find contact by first name\n"
	"5 -> Find contact by last name\n"
	"6 -> Edit contact\n"
	"7 -> Back to menu\n")

	select = int(input("Enter preferred number: "))	
	match select:
		case 1: add_contact()
		case 2: "remove_contact"
		case 3: findcontact_byphonenumber()
		case 4: findcontact_byfisrtname()
		case 5: findcontact_bylastname()
		case 6: edit_contact()
		case 7: menu()
	


print("CHOOSE PERFERRED OPTION\n"
	"1 -> Add contact\n"
	"2 -> Remove contact\n"
	"3 -> Find contact by phone number\n"
	"4 -> Find contact by first name\n"
	"5 -> Find contact by last name\n"
	"6 -> Edit contact\n"
	"7 -> Back to menu\n")

select = int(input("Enter preferred number: "))	
match select:
	case 1: add_contact()
	case 2: "remove_contact"
	case 3: findcontact_byphonenumber()
	case 4: findcontact_byfisrtname
	case 5: findcontact_bylastname()
	case 6: edit_contact()
	case 7: menu()